crates/server/src/authority/catalog.rs

```diff
@@ -47,22 +47,23 @@ async fn send_response<'a, R: ResponseHandler>(
     >,
     mut response_handle: R,
 ) -> io::Result<ResponseInfo> {
-    #[cfg(feature = "dnssec")]
     if let Some(mut resp_edns) = response_edns {
-        // set edns DAU and DHU
-        // send along the algorithms which are supported by this authority
-        let mut algorithms = SupportedAlgorithms::default();
-        algorithms.set(Algorithm::RSASHA256);
-        algorithms.set(Algorithm::ECDSAP256SHA256);
-        algorithms.set(Algorithm::ECDSAP384SHA384);
-        algorithms.set(Algorithm::ED25519);
-
-        let dau = EdnsOption::DAU(algorithms);
-        let dhu = EdnsOption::DHU(algorithms);
-
-        resp_edns.options_mut().insert(dau);
-        resp_edns.options_mut().insert(dhu);
-
+        #[cfg(feature = "dnssec")]
```

Comment:
FYI I think this was a bug. We should be setting the response EDNS here regardless of whether dnssec is enabled. The new code needs this so that it can later correctly determine the limit to use for truncation.

Reply:
hm, I don't think so. if dense isn't enabled, then none of these Algorithms will be supported, which is why that is flagged as off. the EDNS record itself is still added.

Reply:
Right that's my change .. previously the EDNS record wasn't being added if dnssec was disabled. Unless I'm missing something.

Reply:
ah, thanks for explaining. I think I misread the diff.